ts/pages/staking/wizard/wizard.tsx

```diff
@@ -56,33 +51,28 @@ const Container = styled.div`
 export const StakingWizard: React.FC<StakingWizardProps> = props => {
     // If coming from the market maker page, poolId will be provided
     const { poolId } = useQuery<{ poolId: string | undefined }>();
-    const { providerState } = props;
+    const { connector, account, chainId } = useWeb3React<Web3Wrapper>();
 
     const dispatch = useDispatch();
     const dispatcher = new Dispatcher(dispatch);
 
-    const networkId = useSelector((state: State) => state.networkId);
-    const apiClient = useAPIClient(networkId);
+    const apiClient = useAPIClient(chainId);
 
     const [stakingPools, setStakingPools] = useState<PoolWithStats[] | undefined>(undefined);
     const [selectedStakingPools, setSelectedStakingPools] = React.useState<UserStakingChoice[] | undefined>(undefined);
     const [currentEpochStats, setCurrentEpochStats] = useState<Epoch | undefined>(undefined);
     const [nextEpochStats, setNextEpochStats] = useState<Epoch | undefined>(undefined);
     const [allTimeStats, setAllTimeStats] = useState<AllTimeStats | undefined>(undefined);
+    const [zrxBalance, setZrxBalance] = useState<BigNumber | undefined>(undefined);
+    const [zrxBalanceBaseUnitAmount, setZrxBalanceBaseUnitAmount] = useState<BigNumber | undefined>(undefined);
 
-    const stake = useStake(networkId, providerState);
+    const stake = useStake(chainId, { account, connector });
     const allowance = useAllowance();
 
-    const allowanceBaseUnits = (providerState.account as AccountReady).zrxAllowanceBaseUnitAmount || new BigNumber(0);
+    const allowanceBaseUnits = ({ address: account } as AccountReady).zrxAllowanceBaseUnitAmount || new BigNumber(0);
```

Comment:
I don't think this is correct either, see above comment. Will need to pull `zrxAllowanceBaseUnitAmount` from elsewhere, unless I'm misunderstanding something. `address` is just a string, right?

Reply:
While you're in there, can you remove the unneeded cast now (`... as AccountReady`) -- makes it easy to miss these things